### ISSUE TITLE
Deactivated matrix accounts should be automatically !quit from IRC

### ISSUE THREAD

**Issue by csoriano1618** [matrix-org/matrix-appservice-irc#514]
In irc.gnome.org (aka GIMPNet) I deactivated my matrix account, hoping that the irc nickname associated with it would disappear from the network. However that didn't happen, and there is no way to login again to change it to a different irc nickname.
So I believe the bug is that once the account is deactivated the nick doesn't log out.

**Comment by Mikaela** [comment]
Deactivating account simply means changing the password into invalid one so that the account cannot be logged in ever again. I think that your deactivated account wil get disconnected automatically within a month  as the idle dicsonnection script is ran the next time.

Otherwise I think this might be a duplicate of either https://github.com/matrix-org/matrix-appservice-irc/issues/484 or https://github.com/matrix-org/matrix-appservice-irc/issues/195#issuecomment-268198021.

**Comment by csoriano1618** [comment]
Yes that probably is. Feel free to close this one in favor of one of those two.

**Comment by ara4n** [comment]
reopening as you shouldn't have to wait a month for it to idle out.

**Comment by Mikaela** [comment]
Can the bridge know if an account has been deactivated, especially from other homeservers?

**Comment by Half-Shot** [comment]
No, but we shouldn't need to. The homeserver *now* leaves you from all rooms when you deactivate which means we will knife the connection when you leave the last bridged room.

**Comment by Half-Shot** [comment]
>which means we will knife the connection when you leave the last bridged room.

We don't, which seems like a distinctly problematic thing. I assume it's because otherwise you won't be contactable via PM but it also means we will have connections around *forever*.

**Comment by Half-Shot** [comment]
We do now have the ability to deactivate users in the bridge, but there is no way to detect a deactivation from a bridge until https://github.com/matrix-org/matrix-doc/pull/2438 lands.
